### ISSUE TITLE
react-dom: npm install --production error

### ISSUE THREAD

**Issue by voronianski** [facebook/react#5918]
For some specific reason I need to bundle react-dom with browserify. However I faced a problem with installing its' dependencies. Check my steps:

- download react-dom `.tar` file from npm registry (http://registry.npmjs.org/) and unpack it
- go to the react-dom folder and run `npm install --production` 

This results in an error:

```bash
react-dom dmitri$ npm install --production
npm ERR! Darwin 15.3.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "--production"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3

npm ERR! Cannot read property 'react' of undefined
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/dmitri/github/browserify-cdn-test/node_modules/react-dom/npm-debug.log
```

Interesting fact - if run `npm install` without `--production` flag it gets peer dependencies without errors.

Though if it's issue with peer deps why it returns SO weird error **`npm ERR! Cannot read property 'react' of undefined`**. 

Maybe it's related to react-dom packaging?

If you ask why I need this, check this url - https://wzrd.in/standalone/react-dom@latest and https://wzrd.in/ service.

**Comment by voronianski** [comment]
some details and comments are also available here - http://stackoverflow.com/questions/35002066/npm-install-with-production-option-and-peer-dependencies

**Comment by zpao** [comment]
Looks like an npm bug. Stepping through their code, they're referencing `data.dependencies` but if I log, that's not there for react-dom in with the --production flag (https://github.com/npm/npm/blob/2.x/lib/install.js#L180). It looks like they will only fill in `data.dependencies` with an empty object in dev mode (https://github.com/npm/npm/blob/2.x/lib/install.js#L328) - and `install()` explicitly sets `dev: false` when in production (a few lines above the first link).

So there's 2 things that should happen
1. We can add `dependencies: {}` to our package.json
2. Somebody can fix npm's bug. I don't know if it's present in npm3 (haven't looked)

**Comment by zpao** [comment]
Looks like https://github.com/npm/npm/issues/6581 might cover it (but is missing the critical piece that this only happens when `dependencies` is missing from `package.json`

**Comment by voronianski** [comment]
@zpao it will be super helpful it you'll add `dependencies: {}` to your package.json

>  I don't know if it's present in npm3 (haven't looked)

It looks like in npm@3 there's no such bug - http://stackoverflow.com/questions/35002066/npm-install-with-production-option-and-peer-dependencies?noredirect=1#comment57735270_35002066

Thanks!

**Comment by voronianski** [comment]
@zpao do you consider to release a quick patch for `0.14.x`?

**Comment by zpao** [comment]
The PR is marked for 0.14.x so will get picked up if we do another release on the branch. I think we will do that but no hard timeline for that.

**Comment by zpao** [comment]
FYI: Shipped 0.14.7 with the fix so should be good now.

**Comment by voronianski** [comment]
@zpao thanks a lot!
